tests/syscall/mount/mount-immutables.bats

```diff
@@ -0,0 +1,596 @@
+
+  for m in ${immutable_file_mounts}; do
+    # Skip /proc and /sys since these are special mounts (we have dedicated
+    # tests that cover unmounting ops).
+    if [[ ${m} =~ "/proc" ]] || [[ ${m} =~ "/proc/*" ]] ||
+        [[ ${m} =~ "/sys" ]] || [[ ${m} =~ "/sys/*" ]]; then
+      continue
+    fi
+
+    #
+    docker exec ${syscont} sh -c "mount -o bind /dev/null ${m}"
+    [ "$status" -eq 0 ]
+
+    docker exec ${syscont} sh -c "umount ${m}"
+    [ "$status" -eq 0 ]
+
+    docker exec ${syscont} sh -c "umount ${m}"
+    [ "$status" -ne 0 ]
+
+    #
+    docker exec ${syscont} sh -c "touch ${m}2"
+    [ "$status" -eq 0 ]
+
+    docker exec ${syscont} sh -c "mount -o bind ${m} ${m}2"
+    [ "$status" -eq 0 ]
+
+    docker exec ${syscont} sh -c "umount ${m}2"
+    [ "$status" -eq 0 ]
+
+    docker exec ${syscont} sh -c "umount ${m}"
+    [ "$status" -ne 0 ]
+
+  done
+
+  docker_stop ${syscont}
+}
+
+# Testcase #12.
+@test "immutable null-new mount in inner mnt ns" {
+  if [[ $skipTest -eq 1 ]]; then
+    skip
+  fi
+  local syscont=$(docker_run --rm -v /dev/null:/usr/bin/dpkg-maintscript-helper ${CTR_IMG_REPO}/ubuntu:latest tail -f /dev/null)
+  local immutable_char_mounts=$(list_container_char_mounts ${syscont} "0" "/")
+  run empty_list ${immutable_char_mounts}
+  [ "$status" -ne 0 ]  
+
+  for m in ${immutable_char_mounts}; do
+    # Skip /proc and /sys since these are special mounts (we have dedicated
+    # tests that cover unmounting ops).
+    if [[ ${m} =~ "/proc" ]] || [[ ${m} =~ "/proc/*" ]] ||
+        [[ ${m} =~ "/sys" ]] || [[ ${m} =~ "/sys/*" ]]; then
+      continue
+    fi
+
+    #
+    docker exec ${syscont} sh -c "mount -o bind /dev/null ${m}"
+    [ "$status" -eq 0 ]
+
+    docker exec ${syscont} sh -c "umount ${m}"
+    [ "$status" -eq 0 ]
+
+    docker exec ${syscont} sh -c "umount ${m}"
+    [ "$status" -ne 0 ]
+
+    #
+    docker exec ${syscont} sh -c "touch ${m}2"
+    [ "$status" -eq 0 ]
+
+    docker exec ${syscont} sh -c "mount -o bind ${m} ${m}2"
+    [ "$status" -eq 0 ]
+
+    docker exec ${syscont} sh -c "umount ${m}2"
+    [ "$status" -eq 0 ]
+
+    docker exec ${syscont} sh -c "umount ${m}"
+    [ "$status" -ne 0 ]
+
+  done
+
+  docker_stop ${syscont}
+}
+
+# Testcase #13.
+@test "immutable tmpfs mount in inner mnt ns" {
+  if [[ $skipTest -eq 1 ]]; then
+    skip
+  fi
+  local syscont=$(docker_run --rm --mount type=tmpfs,destination=/app ${CTR_IMG_REPO}/ubuntu:latest tail -f /dev/null)
+  local immutable_dir_mounts=$(list_container_dir_mounts ${syscont} "0" "/")
+  run empty_list ${immutable_dir_mounts}
+  [ "$status" -ne 0 ]
+
+  for m in ${mmutable_dir_mounts}; do
+    # Skip /proc and /sys since these are special mounts (we have dedicated
+    # tests that cover unmounting ops).
+    if [[ ${m} =~ "/proc" ]] || [[ ${m} =~ "/proc/*" ]] ||
+        [[ ${m} =~ "/sys" ]] || [[ ${m} =~ "/sys/*" ]]; then
+      continue
+    fi
+
+    # Create bind-mount and verify that only the first unmount is allowed.
```

Comment:
Comment is incorrect, these are not bind mounts (they are overlapping / stacked mounts)